### PR TITLE
Correct change ObjectCreationExpression to ImplictObjectCreationExpression for SmartBreakline

### PR DESCRIPTION
**Pull request by Cosifne**
Fix [AB#1484764](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1484764)
It is caused because
```
Bar b = new
```
`new` is considered as ObjectCreationExpressionSyntax, with the TypeSyntax node missing.
The existing code will only move the trailing trivia of the TypeSyntax after the newly generated argument list, and because the TypeSyntax is missing, it would become
```
Bar b = new
()
{
}
```
The fix is to check if ObjectCreationExpressionSyntax doesn't have a TypeSyntax, then generate a ImplicitObjectCreationExpressionSyntax.

### CODE REVIEW

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs [hunk 1]
```diff
@@ -491,7 +517,7 @@ private static bool ShouldAddBraceForLocalFunctionStatement(LocalFunctionStateme
         /// Add brace for ObjectCreationExpression if it doesn't have initializer
         /// </summary>
         private static bool ShouldAddBraceForObjectCreationExpression(ObjectCreationExpressionSyntax objectCreationExpressionNode)
-            => objectCreationExpressionNode.Initializer == null;
+            => objectCreationExpressionNode.Initializer == null && !objectCreationExpressionNode.NewKeyword.IsMissing;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs:520]: if the NewKeyword is missing... i'm not sure what state we're in :)

Comment (Cosifne) [reply src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs:521]: so it sounds like a unnecessary check? I am little bit afraid of if the 'new' keyword is missing but if the parser can make sure it exist then this is not needed

Comment (Cosifne) [reply src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs]: OK, I removed this.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs]: basically, if we don't have a `new` then the parser should have never gotten into even making an object-creeation in the first place.  THat said, to know that, you have to know how the parser is built.  so it's like 50/50 on if the check should stay or not.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**